open-api/rest-catalog-open-api.yaml

```diff
@@ -3044,29 +3088,51 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/PlanTask'
-        plan-status:
-            $ref: '#/components/schemas/PlanStatus'
 
+    EmptyResult:
+      type: object
+      description: Empty object result
 
-    RetrieveTasksResult:
+    PlanningResult:
       type: object
-      description: Used to fetch file scan tasks for a given `planTask`. Can also return additional plan-tasks.
+      description: Result of server-side scan planning
+      discriminator:
+        propertyName: status
+        mapping:
+          completed: '#/components/schemas/ScanTasks'
+          submitted: '#/components/schemas/EmptyResult'
+          cancelled: '#/components/schemas/EmptyResult'
+          failed: '#/components/schemas/IcebergErrorResponse'
+      oneOf:
+        - $ref: '#/components/schemas/ScanTasks'
+        - $ref: '#/components/schemas/IcebergErrorResponse'
+        - $ref: '#/components/schemas/EmptyResult'
+      required:
+        - status
       properties:
-        file-scan-tasks:
-          type: array
-          items:
-            $ref: '#/components/schemas/FileScanTask'
-        plan-tasks:
-          type: array
-          items:
-            $ref: '#/components/schemas/PlanTask'
+        status:
+          $ref: '#/components/schemas/PlanStatus'
 
-    CancelPlanResult:
+    PlanStatus:
+      description: Status of a server-side planning operation
+      type: string
+      enum: [completed, submitted, cancelled, failed]
+
+    PlanTableScanResult:
       type: object
-      description: Used to indicate state of cancellation. If successful should return "cancelled" state.
+      description: Response schema for planTableScan
+      allOf:
+        - $ref: '#/components/schemas/PlanningResult'
       properties:
-        cancel-status:
-          $ref: '#/components/schemas/PlanStatus'
+        plan-id:
+          description: ID used to track a planning request
+          type: string
+
+    FetchScanTasksResult:
+      type: object
+      description: Response schema for fetchScanTasks
+      allOf:
+        - $ref: '#/components/schemas/ScanTasks'
```

Comment:
The schema here is just the tasks, unlike the `planTableScan` (`POST`) and `fetchPlanningResult` (`GET`) endpoints.